### ISSUE TITLE
Investigate TestSecondStepAfterFailure failing on CI sometimes

### ISSUE THREAD

**Issue by devtomnl** [dusk-network/dusk-blockchain#565]
```
=== RUN   TestSecondStepAfterFailure

==================

WARNING: DATA RACE

Read at 0x00c0001203d8 by goroutine 55:

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep.TestSecondStepAfterFailure()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep/reduction_test.go:61 +0x31a

  testing.tRunner()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:909 +0x199

Previous write at 0x00c0001203d8 by goroutine 91:

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep.(*Reducer).Halt()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep/reducer.go:160 +0x92e

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep.(*Reducer).Halt-fm()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep/reducer.go:148 +0x7d

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep.(*aggregator).collectVote()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep/aggregator.go:76 +0x892

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep.(*Reducer).Collect()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep/reducer.go:111 +0x8d1

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction.(*Helper).SendBatch.func1()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/testutil.go:103 +0xa5

Goroutine 55 (running) created at:

  testing.(*T).Run()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:960 +0x651

  testing.runTests.func1()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:1202 +0xa6

  testing.tRunner()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:909 +0x199

  testing.runTests()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:1200 +0x521

  testing.(*M).Run()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:1117 +0x2ff

  main.main()

      _testmain.go:102 +0x337

Goroutine 91 (finished) created at:

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction.(*Helper).SendBatch()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/testutil.go:102 +0x1b1

  github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep.TestSecondStepAfterFailure()

      /home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep/reduction_test.go:50 +0x155

  testing.tRunner()

      /home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:909 +0x199

==================

--- FAIL: TestSecondStepAfterFailure (2.08s)

    reduction_test.go:61: 

        	Error Trace:	reduction_test.go:61

        	Error:      	Not equal: 

        	            	expected: 2000000000

        	            	actual  : 4000000000

        	Test:       	TestSecondStepAfterFailure

    testing.go:853: race detected during execution of test

FAIL

coverage: 88.2% of statements

FAIL	github.com/dusk-network/dusk-blockchain/pkg/core/consensus/reduction/secondstep	3.987s
```

https://travis-ci.org/github/dusk-network/dusk-blockchain/jobs/699219538

**Comment by ggoranov** [comment]
This issue should also address the race condition detected on running  `TestSecondStep`.

**Comment by julesdesmit** [comment]
It appears that the race (and the subsequent failure) is caused due to the test timer and the `timeOut` being the same length. In no other scenario should any of the node components have direct access to this field of the `reducer`, so guarding it would be unnecessary. A simple adjustment of the reducer timeout will do.

**Comment by julesdesmit** [comment]
Fixed via #602
